pkg/apis/v1beta1/nodepool_validation_cel_test.go

```diff
@@ -0,0 +1,97 @@
+
+	"github.com/Pallinder/go-randomdata"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/samber/lo"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/aws/karpenter-core/pkg/apis"
+	. "github.com/aws/karpenter-core/pkg/apis/v1beta1"
+	"github.com/aws/karpenter-core/pkg/operator/scheme"
+	"github.com/aws/karpenter-core/pkg/test"
+)
+
+var _ = Describe("CEL/Validation", func() {
+	var nodePool *NodePool
+	var env *test.Environment
+
+	BeforeEach(func() {
+		env = test.NewEnvironment(scheme.Scheme, test.WithCRDs(apis.CRDs...))
+		nodePool = &NodePool{
+			ObjectMeta: metav1.ObjectMeta{Name: strings.ToLower(randomdata.SillyName())},
+			Spec: NodePoolSpec{
+				Template: NodeClaimTemplate{
+					Spec: NodeClaimSpec{
+						NodeClassRef: &NodeClassReference{
+							Kind: "NodeClaim",
+							Name: "default",
+						},
+						Requirements: []v1.NodeSelectorRequirement{
+							{
+								Key:      CapacityTypeLabelKey,
+								Operator: v1.NodeSelectorOpExists,
+							},
+						},
+					},
+				},
+			},
+		}
+	})
+	Context("Disruption", func() {
+		It("should fail on negative expireAfter", func() {
+			nodePool.Spec.Disruption.ExpireAfter.Duration = lo.ToPtr(lo.Must(time.ParseDuration("-1s")))
+			Expect(env.Client.Create(ctx, nodePool)).ToNot(Succeed())
+		})
+		It("should succeed on a disabled expireAfter", func() {
+			nodePool.Spec.Disruption.ExpireAfter.Duration = nil
+			Expect(env.Client.Create(ctx, nodePool)).To(Succeed())
+		})
+		It("should succeed on a valid expireAfter", func() {
+			nodePool.Spec.Disruption.ExpireAfter.Duration = lo.ToPtr(lo.Must(time.ParseDuration("30s")))
+			Expect(env.Client.Create(ctx, nodePool)).To(Succeed())
+		})
+		It("should fail on negative consolidateAfter", func() {
+			nodePool.Spec.Disruption.ConsolidateAfter = &NillableDuration{Duration: lo.ToPtr(lo.Must(time.ParseDuration("-1s")))}
+			Expect(env.Client.Create(ctx, nodePool)).ToNot(Succeed())
+		})
+		It("should succeed on a disabled consolidateAfter", func() {
+			nodePool.Spec.Disruption.ConsolidateAfter = &NillableDuration{Duration: nil}
+			Expect(env.Client.Create(ctx, nodePool)).To(Succeed())
+		})
+		It("should succeed on a valid consolidateAfter", func() {
+			nodePool.Spec.Disruption.ConsolidateAfter = &NillableDuration{Duration: lo.ToPtr(lo.Must(time.ParseDuration("30s")))}
+			nodePool.Spec.Disruption.ConsolidationPolicy = ConsolidationPolicyWhenEmpty
+			Expect(env.Client.Create(ctx, nodePool)).To(Succeed())
+		})
+		It("should succeed when setting consolidateAfter with consolidationPolicy=WhenEmpty", func() {
+			nodePool.Spec.Disruption.ConsolidateAfter = &NillableDuration{Duration: lo.ToPtr(lo.Must(time.ParseDuration("30s")))}
+			nodePool.Spec.Disruption.ConsolidationPolicy = ConsolidationPolicyWhenEmpty
+			Expect(env.Client.Create(ctx, nodePool)).To(Succeed())
+		})
+		It("should fail when setting consolidateAfter with consolidationPolicy=WhenUnderutilized", func() {
```

Comment:
Can we also have a check that validates that `consolidationPolicy=WhenUnderutilized` works with `consolidateAfter=Never`